rust/src/shard.rs

```diff
@@ -446,3 +446,30 @@ pub fn create<T: Default + Shard>() -> ShardWrapper<T> {
     error: None,
   }
 }
+
+macro_rules! decl_activate_override {
```

Comment:
I won't be consistent with English grammar though. I like that a macro reads as a sentence: "declare activate's override" and "implement activate's override".

Reply:
I like having this one be more like a verb since it actually replaces the override function to call, the other macro just defines some functions. Up to you